### PR TITLE
[Deepseek V3.2] Change indexer weights_proj to fp32

### PR DESCRIPTION
**Pull request by hlu1**
## Motivation

Following https://github.com/deepseek-ai/DeepSeek-V3.2-Exp/commit/8631a813356d39b09a5c1ee1cde8ed6015559eb3, change the indexer `weights_proj` precision from bf16 to fp32. 

https://github.com/sgl-project/sglang/pull/13439 fixed the rope issue.

## Modifications
- change the indexer `weights_proj` precision from bf16 to fp32
- The `fuse_wk_and_weights_proj` optimization is no longer valid after the fix because the `wk` precision is bf16 and `weights_proj` precision is fp32. Remove it from the code base.
- Add aime 2025 accuracy results and commands to the dsv32 doc. Also added gsm8k 20 shots results. Accuracy results are tested together with https://github.com/sgl-project/sglang/pull/13439 on top of commit 6448b4cd2c4919a06a7d41989abdf435599dd80f

## Accuracy Tests

See updated doc in PR

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/layers/attention/nsa/nsa_indexer.py [hunk 1]
```diff
@@ -808,8 +782,7 @@ def forward_cuda(
             index_k_scale=k_scale,
         )
 
-        if not self.fuse_wk_and_weights_proj:
-            weights, _ = self.weights_proj(x)
+        weights, _ = self.weights_proj(x.float())
```
Comment (trevor-m) [inline python/sglang/srt/layers/attention/nsa/nsa_indexer.py]: I believe this gemm was orignally in the `_get_logits_head_gate()` function.

Comment (hlu1) [reply python/sglang/srt/layers/attention/nsa/nsa_indexer.py]: Moved it back

**Review by trevor-m — APPROVED** [review]

**Review by Fridge003 — APPROVED** [review]
